astropy/nddata/nddata.py

```diff
@@ -180,10 +178,25 @@ def __repr__(self):
 
     @property
     def data(self):
+        """
+        `~numpy.ndarray`: the data
+
+        Even though it is only enforced that the `data` is `~numpy.ndarray`
```

Comment:
I think that may be a residual of when I wanted to move some of the `init` logic to specific setters. And the text defines how the setter originally worked (*must be an ndarray* because overwriting the data with something else would be ambiguous). Should I delete the whole text after 'Even though ...'  in that docstring?